src/frontend/src/components/Layout/Layout.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import Header from '../Header';
+import Banner from '../Banner';
+import Posts from '../Posts';
+import ScrollToTop from '../ScrollToTop';
+import SEO from '../SEO';
+import useSiteMetaData from '../../hooks/use-site-metadata';
+
+function Layout() {
+  const [numPages, setNumPages] = useState(1);
+  const [posts, setPosts] = useState([]);
+  const { telescopeUrl } = useSiteMetaData();
+
+  useEffect(() => {
+    async function getPosts(pageNum = 1) {
+      let postsData = [];
+      try {
+        // should be using telescopeUrl here
+        const res = await fetch(`${telescopeUrl}/posts?page=${pageNum}`);
+        const postsUrls = await res.json();
+        postsData = await Promise.all(
+          postsUrls.map(async ({ url }) => {
+            // should be using telescopeUrl here
+            const tmp = await fetch(`${telescopeUrl}${url}`);
+            const post = await tmp.json();
+            return post;
+          })
+        );
+      } catch (err) {
+        console.log({ err });
+      }
+
+      setPosts(...posts, postsData);
+      setNumPages(numPages + 1);
+    }
+
+    getPosts(numPages);
+  }, [telescopeUrl]);
```

Comment:
It looks like you rely on `numPages` as well.

Reply:
I thought the same, it crashed when I try this

Reply:
Interesting.  OK, let's leave it then.  Seems odd.